WorkloadManagementSystem/Utilities/RabbitMQSynchronizer.py

```diff
@@ -0,0 +1,112 @@
+    warmUp()
+    self._accessUserGroup = 'lhcb_pilot'  #only users belonging to group with this property are allowed to connect
+    self._accessProperty = 'GenericPilot' #only host with this property are allowed to connect
+
+  def sync( self, _eventName, _params ):
+    '''Synchronizes the internal RabbitMQ user database with the current content of CS.
+
+    Args:
+      _eventName: any value, this parameter is ignored, but needed by caller function.
+      _params: - any value, this parameter is ignored, but needed by caller function.
+    Returns:
+      S_OK:
+    Example:
+       s.sync( None, None )
+    '''
+
+    valid_users = getDNsInGroup(self._accessUserGroup)
+    valid_hosts = getDNsForValidHosts(self._accessProperty)
+    updateRabbitMQDatabase(valid_users + valid_hosts)
+    return S_OK()
+
+def getDNsForValidHosts(accessProperty):
+  """Returns DN of hosts which contains accessProperty based on current
+     CS settings.
+  Args:
+    accessProperty(str):
+  Returns:
+    list: of hosts with accessProperty set.
+  """
+
+  retVal = getHosts()
+  if not retVal[ 'OK' ]:
+    return []
+  hosts = retVal['Value']
+  DNs = []
+  for host in hosts:
+    if hostHasProperties(host, [accessProperty]):
+      retVal = getDNForHost(host)
+      if retVal[ 'OK' ]:
+        DNs.extend(retVal['Value'])
+      else:
+        print 'Could not find a correct DN for host: %s. It will be ignored.'% host
+  return DNs
+
+
+def updateRabbitMQDatabase(newUsers, specialUsers = None):
+  """Updates the internal user database of RabbitMQ server.
+     The current user list from the database is compared to newUsers list.
+     The users that are not present in the database, but are in newUsers list
+     are added. The users that are present in the database, but are not in
+     newUsers list are deleted. The specialUser list contains logins that
+     will not be processed at all.
+  Args:
+    newUsers(list): user logins to be processed.
+    specialUsers(list): special users that will not be processed.
+  """
+  if specialUsers is None:
+    specialUsers = ['admin', 'ala', 'O=client,CN=kamyk']
+  ret = getAllUsers()
+  if not ret['OK']:
+    print "Some problem with getting all users from RabbitMQ DB"
+    return ret
+  currentUsersInRabbitMQ =  ret['Value']
+  #special users should not be taken into account
+  currentUsersInRabbitMQ = listDifference(currentUsersInRabbitMQ, specialUsers)
+  usersToRemove = listDifference(currentUsersInRabbitMQ, newUsers)
+  usersToAdd = listDifference(newUsers, currentUsersInRabbitMQ)
+  if usersToAdd:
+    addUsersWithoutPasswords(usersToAdd)
+    setUsersPermissions(usersToAdd)
+  if usersToRemove:
+    deleteUsers(usersToRemove)
+  return S_OK()
+
+def listDifference(list1, list2):
+  """Calculates differences between two lists.
+     The original order of the list is not preserved.
+  Returns:
+    list:
+  """
+  return list(set(list1) - set(list2))
+
+def warmUp():
```

Comment:
Right. I will fix it.
 but actually I wanted to ask if it is needed at all? in the one example with the CS sync that you sent me it was used, but It is not used in any other part of the code. The function comment is a bit criptic.